src/sdk.ts

```diff
@@ -74,129 +88,653 @@ export class CoherenceBeamsplitterSDK {
+              transferMint: weightedToken.mint,
+              ordererTransferAta,
+              beamsplitterTransferAta,
+              beamsplitter,
+              rent: SYSVAR_RENT_PUBKEY,
+              associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+              tokenProgram: TOKEN_PROGRAM_ID,
+              systemProgram: SystemProgram.programId,
+            },
+          })
+        )
       );
     }
 
-    return tx.combine(setAuthTx);
+    return constructTxChunks;
   }
 
-  getPrice({
-    owner = this.provider.wallet.publicKey,
-    price,
-    priceSigner,
-    market,
-    bids,
-    dexPid = this.getLatestSerumDEXAddress(),
+  async decohere({
+    beamsplitter,
+    prismEtfMint,
   }: {
-    owner?: PublicKey;
-    price: PublicKey;
-    priceSigner: Signer;
-    market: PublicKey;
-    bids: PublicKey;
-    dexPid?: PublicKey;
-  }): TransactionEnvelope {
-    return new TransactionEnvelope(
-      this.provider,
-      [
-        this.program.instruction.getPrice(dexPid, {
-          accounts: {
-            price,
-            payer: owner,
-            systemProgram: SystemProgram.programId,
-          },
-          remainingAccounts: [
-            {
-              pubkey: market,
-              isWritable: false,
-              isSigner: false,
-            },
-            {
-              pubkey: bids,
-              isWritable: false,
-              isSigner: false,
+    beamsplitter: PublicKey;
+    prismEtfMint: PublicKey;
+  }): Promise<TransactionEnvelope[]> {
+    const orderState = await this.fetchOrderStateDataFromSeeds({
+      beamsplitter,
+      prismEtfMint,
+    });
+
+    if (!orderState) {
+      throw new Error(
+        "You must start an order first. Call startOrder() before construct."
+      );
+    }
+
+    const prismEtf = await this.fetchPrismEtfDataFromSeeds({
+      beamsplitter,
+      prismEtfMint,
+    });
+
+    if (!prismEtf) {
+      throw new Error("You must create the prismEtf first.");
+    }
+
+    const weightedTokensAcct = await this.fetchWeightedTokens(
+      prismEtf.weightedTokens
+    );
+
+    if (!weightedTokensAcct) {
+      throw new Error("Weighted tokens was not initalized.");
+    }
+
+    const weightedTokens = weightedTokensAcct.weightedTokens;
+
+    const [prismEtfPda] = await generatePrismEtfAddress(
+      prismEtfMint,
+      beamsplitter
+    );
+
+    const [orderStatePda] = await generateOrderStateAddress(
+      prismEtfMint,
+      beamsplitter,
+      this.provider.wallet.publicKey
+    );
+
+    const weightedTokensActualLength = weightedTokensAcct.index;
+    const constructTxChunks: TransactionEnvelope[] = [];
+    for (let i = 0; i < weightedTokensActualLength; i++) {
+      const constructEnvelope = new TransactionEnvelope(this.provider, []);
+
+      if (!weightedTokens.at(i)) {
+        throw new Error("Outside weighted tokens array range");
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Safe to disable this?

Reply:
Basically client side it doesn't know with confidence how long the array is because it's being serialized at runtime. Don't think we can have this typed, so have to use this here.